plugins/services/src/js/data/ServicePlansModel.ts

```diff
@@ -0,0 +1,123 @@
+import { Observable } from "rxjs";
+import { RequestResponse } from "@dcos/http-service";
+import {
+  fetchPlans as fetchServicePlans,
+  fetchPlanDetails as fetchServicePlanDetail,
+  ServicePlanResponse
+} from "#PLUGINS/services/src/js/data/ServicePlansClient";
+import { makeExecutableSchema } from "graphql-tools";
+
+import Config from "#SRC/js/config/Config";
+import { ServicePlanStatusSchema } from "#PLUGINS/services/src/js/structs/ServicePlanStatus";
+import { ServicePlanStepSchema } from "#PLUGINS/services/src/js/structs/ServicePlanStep";
+import { ServicePlanPhaseSchema } from "#PLUGINS/services/src/js/structs/ServicePlanPhase";
+import {
+  ServicePlan,
+  ServicePlanDetailResolver,
+  ServicePlanSchema
+} from "#PLUGINS/services/src/js/structs/ServicePlan";
+
+export interface Query {
+  servicePlans: string[] | null;
+  servicePlan: ServicePlan | null;
+}
+
+export interface ResolverArgs {
+  fetchServicePlans: (
+    serviceName: string
+  ) => Observable<RequestResponse<string[]>>;
+  fetchServicePlanDetail: (
+    serviceName: string,
+    planName: string
+  ) => Observable<RequestResponse<ServicePlanResponse>>;
+  pollingInterval: number;
+}
+
+export interface GeneralArgs {
+  [key: string]: any;
+}
+
+export interface ServicePlansQueryArgs {
+  serviceName: string;
+}
+
+export interface ServicePlanDetailQueryArgs {
+  serviceName: string;
+  planName: string;
+}
+
+export const typeDefs = `
+${ServicePlanStatusSchema}
+${ServicePlanStepSchema}
+${ServicePlanPhaseSchema}
+${ServicePlanSchema}
+
+type Query {
+  servicePlans(serviceName: String!): [String!]!
```

Comment:
While this is technically correct, we kinda waste the advantages of graphql here by treating it as rest. In an ideal world we had already the service type and as a consumer could do something like this:


```
{
  query {
    service(serviceName: "MyService") {
       plans {
          name
          phases {
             name
          }
       }
    }
  }
}
```

as we don't have services it sounds to me like we should rather do sth like this:


```
{
  query {
    servicePlans(serviceName: "MyService") {
       name
       phases {
          name
       }
    }
  }
}
```